assets/block/src/edit.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Retrieves the translation of text.
+ *
+ * @see https://developer.wordpress.org/block-editor/packages/packages-i18n/
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * React hook that is used to mark the block wrapper element.
+ * It provides all the necessary props like the class name.
+ *
+ * @see https://developer.wordpress.org/block-editor/packages/packages-block-editor/#useBlockProps
+ */
+import { ServerSideRender } from '@wordpress/server-side-render';
+import {
+    useBlockProps,
+    RichText,
+    BlockControls,
+	InspectorControls,
+	AlignmentToolbar
+} from '@wordpress/block-editor';
+import {
+	PanelBody,
+	TextControl,
+	Dashicon,
+	Toolbar,
+	ToolbarButton,
+	Button,
+	Tooltip,
+	SelectControl
+} from '@wordpress/components';
+import { tablet } from '@wordpress/icons';
+
+import classnames from 'classnames';
+
+/**
+ * Lets webpack process CSS, SASS or SCSS files referenced in JavaScript files.
+ * Those files can contain any CSS code that gets applied to the editor.
+ *
+ * @see https://www.npmjs.com/package/@wordpress/scripts#using-css
+ */
+import './editor.scss';
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/developers/block-api/block-edit-save/#edit
+ *
+ * @return {WPElement} Element to render.
+ */
+export default function Edit( props ) {
+	console.log('entered here ' + JSON.stringify(props.attributes));
+	console.log('isselected ' + props.isSelected);
+
+	const blockProps = useBlockProps();
+	/*const {
+		attributes: { slug, contribution_type, preferred_username, theme },
+		className,
+	} = props;
+
+	useEffect = () => {
+
+	};*/
```

Comment:
I believe this PR was issued most just to try and ease with my reviewing? If yes, then I won't stress this part that much, but when it comes to intended to be pulled in, we'll want to clean up this part, as I'm sure you know.

Reply:
I'll be updating the final now, this one update was still with some issues, now it's working well I can cleanup,